### ISSUE TITLE
Multiple databases brainstorming

### ISSUE THREAD

**Issue by ivanovyordan** [transferwise/pipelinewise-tap-mongodb#39]
I stumbled upon an interesting MongoDB design that uses a separate database for each user. All these databases are being used by the same app and have the same structure.

Quick research showed this a common design pattern in the Mongo world. It's supposed to be more performant, provides better logical replication and potentially better security. Example: https://stackoverflow.com/questions/16916903/mongodb-performance-having-multiple-databases

The easiest way to deal with this pattern is to build a script that gets all databases and run the sync for every database one by one. I was wondering though if there's a will into implementing a feature that supports multiple databases within the same pipeline?

If this is to be implemented what would be the best way? I can see only one option for now - Add a `*` sign in the database name. So if we set the database to `user_*`, this would go through all user databases.

**Comment by Samira-El** [comment]
Hi @ivanovyordan, I'm not an expert in MongoDb and don't know if that solution/approach still holds true in 2021 (the SO thread is from 8 years ago).

re supporting multiple databases, that's something we had to forego when we decided to create our own tap-mongodb (inspired by Singer.io https://github.com/singer-io/tap-mongodb) and implement `LOG_BASED` using ChangeStreams. So, adding this feature of multiple dbs would require some serious code changes.
